src/Request.php

```diff
@@ -976,4 +985,70 @@ public static function getWebhookInfo()
         // Must send some arbitrary data for this to work for now...
         return self::send('getWebhookInfo', ['info']);
     }
+
+    /**
+     * Enable request limiter
+     *
+     * @param boolean $value
+     */
+    public static function setLimiter($value = true)
+    {
+        if (DB::isDbConnected()) {
+            self::$limiter_enabled = $value;
+        }
+    }
+
+    /**
+     * This functions delays API requests to prevent reaching Telegram API limits
+     *  Can be disabled while in execution by 'Request::setLimiter(false)'
+     *
+     * @link https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this
+     *
+     * @param string $action
+     * @param array  $data
+     *
+     * @throws \Longman\TelegramBot\Exception\TelegramException
+     */
+    private static function limitTelegramRequests($action, array $data = [])
+    {
+        if (self::$limiter_enabled) {
+            $limited_methods = [
+                'sendMessage',
+                'forwardMessage',
+                'sendPhoto',
+                'sendAudio',
+                'sendDocument',
+                'sendSticker',
+                'sendVideo',
+                'sendVoice',
+                'sendLocation',
+                'sendVenue',
+                'sendContact',
+                'editMessageText',
+                'editMessageCaption',
+                'editMessageReplyMarkup',
+            ];
+
+            if ((isset($data['chat_id']) || isset($data['inline_message_id'])) && in_array($action, $limited_methods)) {
+                $timeout = 60;
+
+                while (true) {
+                    if ($timeout <= 0) {
+                        throw new TelegramException('Timed out while waiting for a request slot!');
+                    }
+
+                    $requests = DB::getTelegramRequestCount((isset($data['chat_id']) ? $data['chat_id'] : null), (isset($data['inline_message_id']) ? $data['inline_message_id'] : null));
+
+                    if ($requests['LIMIT_PER_SEC'] <= 0 && ((isset($data['inline_message_id']) && $requests['LIMIT_PER_SEC_ALL'] < 30) || (isset($data['chat_id']) && $data['chat_id'] > 0 && $requests['LIMIT_PER_SEC_ALL'] < 30) || (isset($data['chat_id']) && $data['chat_id'] < 0 && $requests['LIMIT_PER_MINUTE'] < 20))) {
```

Comment:
For this it would make sense to add variables for `$data['chat_id']` and `$data['inline_message_id']` as they're also used further up.

So instead of all the `isset() && ...` you could just use the variables directly.

Will think about this exact spot and how to make it more readable 👍